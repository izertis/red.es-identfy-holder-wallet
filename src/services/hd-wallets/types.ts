import { derivationPathFormats } from './constants'

type randomType = 'random'

export type settingType = {
	m?: undefined
	Z?: string
	R?: string | randomType
	SSSSS?: [string, string, string, string, string] | randomType
	W?: string | randomType
	M?: string
	T?: string
	N?: string
	B?: randomType
	BB?: [string, string] | randomType
}
export type validKeysType = keyof settingType

export type derivationPathFormat =
	| derivationPathFormats.MAIN_IDENTITY
	| derivationPathFormats.NETWORK_DID
	| derivationPathFormats.INTERACTING_DID
