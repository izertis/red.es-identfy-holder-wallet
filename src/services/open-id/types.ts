import { Jwt, JwtHeader, JwtPayload } from 'jsonwebtoken'
import { ModalPropsWithoutIsVisible } from '../../context/Modal.context'

export type ModalType = {
	showModal?: (props?: ModalPropsWithoutIsVisible) => void
	t?: (s: string) => string
	showMessage?: (x: { content: string; type: string }) => void
}

export type Grants = {
	authorization_code: {
		issuer_state: string
	}
	'urn:ietf:params:oauth:grant-type:pre-authorized_code'?: {
		'pre-authorized_code': string
		user_pin_required: boolean
	}
}

export type PreAuthorizeResponse = {
	access_token: string
	token_type: string
	expires_in: number
	id_token: string
	c_nonce: string
	c_nonce_expires_in: number
}

export type CredentialOffer = {
	credential_issuer: string
	credentials: {
		format: string
		types: string[]
		trust_framework: {
			name: string
			type: string
			uri: string
		}
	}[]
	grants: Grants
}

export type PresentationOffer = {
	aud: string
	client_id: string
	exp: number
	iat: number
	iss: string
	nonce: string
	presentation_definition: PresentationDefinition
	redirect_uri: string
	response_mode: string
	response_type: string
	scope: string
	state?: string
}

export type OpenIdCredentialIssuer = {
	credential_issuer: string
	authorization_server: string
	credential_endpoint: string
	deferred_credential_endpoint: string
	credentials_supported: {
		format: string
		types: string[]
		trust_framework: {
			name: string
			type: string
			uri: string
		}
		display: {
			name: string
			locale: string
		}[]
	}[]
}

export type OpenIdConfiguration = {
	redirect_uris: string[]
	issuer: string
	authorization_endpoint: string
	token_endpoint: string
	jwks_uri: string
	scopes_supported: string[]
	response_types_supported: string[]
	response_modes_supported: string[]
	grant_types_supported: string[]
	subject_types_supported: string[]
	id_token_signing_alg_values_supported: string[]
	request_object_signing_alg_values_supported: string[]
	request_parameter_supported: boolean
	request_uri_parameter_supported: boolean
	token_endpoint_auth_methods_supported: string[]
	request_authentication_methods_supported: {
		[key: string]: string[]
	}
	vp_formats_supported: {
		[key: string]: {
			alg_values_supported: string[]
		}
	}
	subject_syntax_types_supported: string[]
	subject_syntax_types_discriminations: string[]
	subject_trust_frameworks_supported: string[]
	id_token_types_supported: string[]
}

export type CredentialResponse = {
	format: 'jwt_vc'
	credential: string
	acceptance_token: string
}

export type AuthorizationRequest = {
	response_type: 'code'
	client_id: string
	redirect_uri: string
	scope: string
	issuer_state?: string
	state?: string
	authorization_details: string
	nonce?: string
	code_challenge?: string
	code_challenge_method?: 'S256' | 'plain'
	client_metadata?: string
}

export type HolderServiceWalletMetadata = {
	authorization_endpoint: string
	response_types_supported: string[]
	vp_formats_supported: {
		jwt_vp: {
			alg_values_supported: string[]
		}
		jwt_vc: {
			alg_values_supported: string[]
		}
	}
	scopes_supported: string[]
	subject_types_supported: string[]
	id_token_signing_alg_values_supported: string[]
	request_object_signing_alg_values_supported: string[]
	subject_syntax_types_supported: string[]
	id_token_types_supported: string[]
}

export type AuthorizationDetail = {
	type: 'openid_credential'
	locations?: string[]
	format: 'jwt_vp' | 'jwt_vc' | string
	types: string[]
}

export type AuthorizationResponse = {
	code: string
	state?: string
}

export type GrantType = 'urn:ietf:params:oauth:grant-type:pre-authorized_code' | 'preauthorized_code'

export type TokenRequest = {
	redirect_uri: string
	grant_type: GrantType
	client_id?: string
	code?: string
	client_assertion_type?: boolean
	code_verifier?: string
	user_pin?: string
	'pre-authorized_code'?: string
}

export type TokenResponse = {
	access_token: string
	refresh_token: string
	token_type: string
	expires_in: number
	id_token: string
	c_nonce: string
	c_nonce_expires_in: number
}

export type VPTokenRequestPayload = {
	iss: string
	aud: string
	exp: number
	response_type: string
	response_mode: string
	client_id: string
	redirect_uri: string
	scope: string
	nonce: string
	presentation_definition_uri: string
}

export type AuthorizationRequestResponse = {
	client_id: string
	key: string
	nonce: string
	redirect_uri: string
	request_uri: string
	response_mode: string
	response_type: string
	scope: string
	presentation_definition: PresentationDefinition
	presentation_definition_uri: string
	request: string
	error: string
	error_description: string
}

export type DeferredCredential = {
	credential_endpoint: string
	acceptance_token: string
}

export type PresentationDefinition = {
	id: string
	format: Format
	input_descriptors: InputDescriptor[]
	purpose?: string
}

type InputDescriptor = {
	id: string
	name?: string
	format?: Format
	constraints?: {
		fields: FieldConstraint[]
	}
}

type FieldConstraint = {
	path: string[]
	filter: {
		type: string
		contains: {
			const: string
		}
	}
}

export type Format = {
	jwt_vc?: {
		alg: string[]
	}
	jwt_vp?: {
		alg: string[]
	}
	jwt_vc_json?: {
		alg: string[]
	}
	jwt_vp_json?: {
		alg: string[]
	}
}

export type TrustedFramework = 'lacchain' | 'alastria' | 'ebsi'

export type CredentialData = {
	id: string
	issuer: string
	validFrom: any
	expirationDate: any
	timestamp?: Date
	credential: Credential
	status: 'Active' | 'Revoked'
}

// CREDENTIALS SCHEMA
export type Vc = {
	'@context': string[]
	id: string
	type: string[]
	credentialSchema: {
		id: string
		type: string
	}
	credentialSubject: {
		id: string
	}
	validFrom?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	validUntil?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	expirationDate?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	issuanceDate: string
	issuer: string
	issued?: string
	credentialStatus?: W3CCredentialStatus
	proof?: EmbeddedProof
	[x: string]: any
}

export interface EmbeddedProof {
	id?: string
	type: string
	proofPurpose: string
	verificationMethod: string
	created?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	expires?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	domain?: string
	challenge?: string
	proofValue: string
	previousProof?: string | string[]
	nonce?: string
}

export interface W3CCredentialStatus {
	id?: string
	type: string
	[key: string]: any
}

export interface StatusListEntry2021 {
	id: string
	type: 'StatusList2021Entry'
	statusPurpose: 'revocation' | 'suspension'
	statusListIndex: string
	statusListCredential: string
}

export interface StatusListCredentialData {
	id: string
	type: 'StatusList2021'
	statusPurpose: 'revocation' | 'suspension'
	encodedList: string
}

export type Credential = {
	header: JwtHeader
	payload: JwtPayload & {
		vc: Vc
	}
}

export type chosenCredential = {
	format: 'jwt_vc' | 'ldp_vc'
	trust_framework: { name: string; type: string; uri: string }
	types: ['VerifiableCredential', 'VerifiableAttestation', string]
}

export type PresentationData = {
	id: string
	issuer: string
	validFrom: any
	expirationDate: any
	timestamp: Date
	credential: Credential
	status: 'Active' | 'Revoked'
}

export type ActionLog = {
	id: string
	actionType: 'Credential added' | 'Credential expired' | 'Presentation emitted' | 'Presentation revoked'
	issuer: string
	date: Date | string
	data?: any
}
