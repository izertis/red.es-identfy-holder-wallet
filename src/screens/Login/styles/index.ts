import defaultStyles from "./defaultStyles"
import proyectSyles from "./proyectStyles"

const LoginStyled = {
  ...defaultStyles,
  ...proyectSyles
}

export default LoginStyled
