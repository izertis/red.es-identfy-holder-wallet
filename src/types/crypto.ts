export type JWT = `ey${string}.ey${string}.${string}`