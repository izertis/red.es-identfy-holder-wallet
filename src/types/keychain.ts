import { ActionLog, CredentialData, DeferredCredential, PresentationData } from '../services/open-id/types'

export type KeyChainData = {
	encryptedPrivateKey?: string | ArrayBuffer
	did?: { lacchain?: string; alastria?: string; ebsi?: string }
	JWKKeys?: { privateKeyJWK: string; publicKeyJWK: string; privateKeyHex: string; publicKeyHex: string }
	signatureKeyPair?: { publicKey: string; privateKey: string }
	credentials?: CredentialData[]
	presentations?: PresentationData[]
	deferred_credentials?: DeferredCredential[]
	actionLogs?: ActionLog[]
}
