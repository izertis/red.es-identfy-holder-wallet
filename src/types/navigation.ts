import { MaterialTopTabScreenProps } from '@react-navigation/material-top-tabs'
import { NativeStackNavigationOptions } from '@react-navigation/native-stack'

export type navigationScreensConfigType = {
	name: string
	component: (props: any) => JSX.Element
	options:
		| NativeStackNavigationOptions
		| MaterialTopTabScreenProps<{}>
		| ((x: any) => NativeStackNavigationOptions | MaterialTopTabScreenProps<{}>)
}
