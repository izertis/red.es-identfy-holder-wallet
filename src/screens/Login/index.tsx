import { useContext, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { RootStackScreenProps } from '../../../types'
import { SCREEN } from '../../constants/screens'
import { MessageContext } from '../../context/UserMessage.context'
import LocalStorageService, {
  STORAGE_KEYS,
} from '../../services/LocalStorage.service'
import { checkPin } from '../../utils/keychain'
import RegisterI18nKeys from './i18n/keys'
import LoginStyled from './styles'
import { View } from '../../components/Themed'
import { executeCallbackFunction } from '../../utils/storeCallback'
import { StatusBar } from 'expo-status-bar'
import { ColorKeys, getThemeColor } from '../../constants/Colors'
import { Keyboard, Platform } from 'react-native'

const Login = ({ route, navigation }: RootStackScreenProps<'Login'>) => {
  const { t } = useTranslation(SCREEN.Login)
  const { callback }: any = route?.params ?? {}
  const { showMessage } = useContext(MessageContext)
  const [isLoading, setIsLoading] = useState<boolean>(false)
  const [pin, setPin] = useState<string>('')

  const login = async () => {
    Keyboard.dismiss()
    const IS_DID_CREATED = await LocalStorageService.getBool(
      STORAGE_KEYS.IS_DID_CREATED
    )
    await LocalStorageService.storeData(STORAGE_KEYS.PIN, pin)
    setPin('')
    if (IS_DID_CREATED) {
      try {
        callback
          ? await executeCallbackFunction(callback.id, callback.params)
          : navigation.replace('Root')
      } catch (error) { }
    } else {
      navigation.replace('NetworkAuth')
    }
    setPin('')
    setIsLoading(false)
  }

  const showError = (): void => {
    const content = t(RegisterI18nKeys.ERROR)
    showMessage({ content, type: 'error' })
    setPin('')
    setIsLoading(false)
  }

  return (
    <LoginStyled.MainContainer>
      <StatusBar backgroundColor='transparent' style='auto' />
      <View>
        <LoginStyled.RegisterButton
          onPress={() => navigation.navigate('Register')}
        >
          {t(RegisterI18nKeys.REGISTER_LINK)}
        </LoginStyled.RegisterButton>

        <LoginStyled.Title testID="title">
          {t(RegisterI18nKeys.TITLE)}
        </LoginStyled.Title>
        <LoginStyled.InputsContainer>
          <LoginStyled.InputStyled
            disabled={isLoading}
            value={pin}
            keyboardType="decimal-pad"
            secureTextEntry
            onChangeText={(value: string) =>
              setPin(value.replace(/[^0-9]/g, ''))
            }
            placeholder={t(RegisterI18nKeys.PIN_LABEL) || ''}
            onSubmitEditing={async () => ((await checkPin(pin)) ? login() : showError())}
            returnKeyType={Platform.OS === 'android' ? 'send' : 'done'}
          />
        </LoginStyled.InputsContainer>
      </View>
      <LoginStyled.ContainerBottom>
        {isLoading ? <LoginStyled.SplashActivityIndicator color={getThemeColor(ColorKeys.text)} size={35} /> : null}
        <LoginStyled.Button
          testID="startButton"
          disabled={pin === '' || isLoading}
          style={{ flexDirection: 'row' }}
          onPress={async () => {
            setIsLoading(true);
            ((await checkPin(pin)) ? login() : showError())
          }}
        >
          {t(RegisterI18nKeys.BUTTON)}
        </LoginStyled.Button>
      </LoginStyled.ContainerBottom>
    </LoginStyled.MainContainer>
  )
}

export default Login
