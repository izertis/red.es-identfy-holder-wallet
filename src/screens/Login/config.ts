import i18next from '../../../i18n.config'
import { SCREEN } from "../../constants/screens"
import { navigationScreensConfigType } from "../../types/navigation"
import es from "./i18n/es"
import Login from '.'

const ScreenName = SCREEN.Login

i18next.addResourceBundle("es", ScreenName, es)

const LoginConfig: navigationScreensConfigType = {
  name: ScreenName,
  component: Login,
  options: { headerShown: false },
}

export default LoginConfig
