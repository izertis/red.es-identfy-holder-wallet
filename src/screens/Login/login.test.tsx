import { NativeStackNavigationProp } from '@react-navigation/native-stack'
import { render } from '@testing-library/react-native'
import Login from '.'
import { RootStackParamList } from '../../../types'

type ScreenNavigationProp = NativeStackNavigationProp<RootStackParamList, 'Login'>

const mockedNavigation: ScreenNavigationProp | any = {
	navigate: jest.fn(),
	route: jest.fn(),
}

describe('Login screen', () => {
	it('should render login screen', () => {
		const result = render(<Login {...mockedNavigation} />)

		expect(result.toJSON()).toMatchSnapshot()
	})
	it('Should renders items correctly', () => {
		const { getAllByRole, getByTestId } = render(<Login {...mockedNavigation} />)

		const title = getByTestId('title')
		const startBtn = getAllByRole('button')
		const recoverButton = getAllByRole('button')

		expect(title).toBeDefined()
		expect(startBtn).toBeDefined()
		expect(recoverButton).toBeDefined()
	})
})
