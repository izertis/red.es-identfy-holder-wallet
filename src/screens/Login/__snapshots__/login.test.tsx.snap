// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen should render login screen 1`] = `
<View
  className="StyledNativeComponent sc-iBYQkv jyizxs"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        className="sc-pyfCe eKlEPk"
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 20,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                Object {
                  "width": 100,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginLeft": 12,
                    "marginRight": -4,
                  },
                  Object {
                    "marginLeft": 16,
                    "marginRight": -16,
                  },
                  Object {
                    "marginLeft": 12,
                    "marginRight": -8,
                  },
                ]
              }
              testID="button-icon-container"
            >
              <Text />
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 16,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#eb5b5b",
                        "fontFamily": "Muli-600",
                      },
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              REGISTER_LINK
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Text
      bold={true}
      className="sc-gswNZR sc-eDvSVe sc-kDvujY emaRpg kONyvx"
      fontFamily="Montserrat"
      style={
        Array [
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          Object {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
      testID="title"
    >
      TITLE
    </Text>
    <View
      className="sc-ipEyDJ ZCQnv"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            undefined,
          ]
        }
      >
        <View
          style={Object {}}
        >
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "rgba(255, 251, 254, 1)",
                  "borderColor": "#363d58",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
                undefined,
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                  },
                  Object {
                    "minHeight": 56,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                className="sc-bcXHqe sc-csuSiG fkiSib kBBgOM"
                containerStyle={
                  Object {
                    "paddingHorizontal": 0,
                  }
                }
                cursorColor="#363d58"
                editable={true}
                errorStyle={
                  Object {
                    "color": "#eb5b5b",
                  }
                }
                inputContainerStyle={
                  Object {
                    "marginTop": 10,
                  }
                }
                inputStyle={
                  Object {
                    "fontSize": 16,
                  }
                }
                keyboardType="decimal-pad"
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="PIN_LABEL"
                placeholderTextColor="#aaa"
                returnKeyType="done"
                secureTextEntry={true}
                selectionColor="#eb5b5b"
                style={
                  Array [
                    Object {
                      "margin": 0,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 48,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "paddingHorizontal": 16,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Object {
                      "marginRight": 56,
                      "paddingRight": 0,
                    },
                    Object {
                      "fontFamily": "Muli",
                      "lineHeight": 25,
                    },
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 24,
                    "justifyContent": "center",
                    "position": "absolute",
                    "width": 24,
                  },
                  Object {
                    "right": 16,
                    "top": 19,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 20,
                    "height": 40,
                    "margin": 0,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "width": 40,
                  }
                }
                testID="right-icon-adornment-container-outer-layer"
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "flex": 1,
                      "overflow": "hidden",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="right-icon-adornment-container"
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": false,
                      }
                    }
                    accessibilityTraits="button"
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      Object {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          Object {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="right-icon-adornment"
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        className="sc-jrcTuL imoTCb"
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 20,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                Object {
                  "width": 160,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 12,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#eb5b5b",
                        "fontFamily": "Muli-600",
                      },
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              RECOVERY
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    className="sc-eDWCr dISCWl"
    disabled={false}
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="startButton-container-outer-layer"
    >
      <View
        className="StyledNativeComponent sc-ftTHYK"
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "flexDirection": "row",
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="startButton-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 20,
              },
            ]
          }
          testID="startButton"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                Object {},
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 12,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#000",
                        "fontFamily": "Muli",
                        "letterSpacing": 0.57,
                      },
                    ],
                  ],
                ]
              }
              testID="startButton-text"
            >
              BUTTON
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
